core/src/main/java/com/linecorp/armeria/common/AbstractHttpRequestBuilder.java

```diff
@@ -411,42 +425,142 @@ private RequestHeaders requestHeaders() {
 
     private String buildPath() {
         checkState(path != null, "path must be set.");
```

Comment:
Could we early return path itself if `pathParams.isEmpty()` and `queryParams == null`?

Reply:
I think we still should raise an exception when a user specified `{param}` in a path and `param` doesn't exist.

Reply:
That makes sense.